### PR TITLE
[VL] Forbid unsupported write using velox

### PR DESCRIPTION
**Pull request by ulysses-you**
## What changes were proposed in this pull request?

This pr does two things:
- Forbid velox write with ctas. It can be supported in Spark3.4
- Forbid velox dynamic partition and bucket write. As mentioned at https://github.com/oap-project/gluten/issues/2042, it would fail if we use dynamic partition write with velox. There is a lot of work need to be done. This pr just throw exception in advance when contains partition or bucket.

## How was this patch tested?

add test

### CODE REVIEW

**Comment by ulysses-you** [comment]
cc @JkSelf @zhouyuan thank you

File: gluten-core/src/main/scala/org/apache/spark/sql/execution/datasources/GlutenColumnarRules.scala [hunk 1]
```diff
@@ -87,10 +83,26 @@ object GlutenColumnarRules {
...
+  // TODO: support ctas in Spark3.4, see https://github.com/apache/spark/pull/39220
+  // TODO: support dynamic partition and bucket write
+  //  1. pull out `Empty2Null` and required ordering to `WriteFilesExec`, see Spark3.4 `V1Writes`
+  //  2. support detect partition value, partition path, bucket value, bucket path at native side,
+  //     see `BaseDynamicPartitionDataWriter`
   def isGlutenInsertInto(cmd: DataWritingCommand): Boolean = {
     cmd match {
-      case command: InsertIntoHadoopFsRelationCommand =>
-        command.fileFormat.isInstanceOf[GlutenParquetFileFormat]
+      case command: CreateDataSourceTableAsSelectCommand =>
```
Comment (JkSelf) [inline gluten-core/src/main/scala/org/apache/spark/sql/execution/datasources/GlutenColumnarRules.scala:93]: Can we fallback these cases to vanilla spark?

Comment (ulysses-you) [reply gluten-core/src/main/scala/org/apache/spark/sql/execution/datasources/GlutenColumnarRules.scala:93]: I'm not sure it's a clear behavoir to user. If people specify format to `velox` then they want a velox format, so I think it should be better just fast fail if we do not support. And in future, we plan to add rule to convert parquet format to velox if we support, then things would go to complex if we support both velox -> parquet and parquet -> velox ...

Comment (JkSelf) [reply gluten-core/src/main/scala/org/apache/spark/sql/execution/datasources/GlutenColumnarRules.scala:93]: Ok. Can you help to update the doc [here ](https://github.com/oap-project/gluten/blob/main/docs/VeloxNotSupport.md#parquet-write)to add the unsupported command?

Comment (ulysses-you) [reply gluten-core/src/main/scala/org/apache/spark/sql/execution/datasources/GlutenColumnarRules.scala:93]: sure, added docs. I refactor the parquet write docs to `parquet write` and `velox parquet write`. The behavior of `velox parquet write` should no affect Spark user, it only affects who specify format to `velox`. So far, `parquet write` only contains hive file format which has been overwritten by velox parquet write.

**Review by JkSelf — APPROVED** [review]

**Pull request merged**